### ISSUE TITLE
xml2js is vulnerable to prototype pollution

### ISSUE THREAD

**Issue by msouther-lisc** [jsforce/jsforce#1318]
`npm audit` is informing me of this vulnerability:

```
xml2js  <=0.4.23
Severity: high
xml2js is vulnerable to prototype pollution  - https://github.com/advisories/GHSA-776f-qx25-q3cc
No fix available
node_modules/xml2js
  jsforce  *
  Depends on vulnerable versions of xml2js
  node_modules/jsforce
```

See https://github.com/advisories/GHSA-776f-qx25-q3cc

There hadn't been an update to xml2js in over 3 years until just yesterday, when a new minor version, 0.5.0, was released:

https://www.npmjs.com/package/xml2js?activeTab=versions

Seems worth updating the dependency.

**Comment by PGN1** [comment]
any thoughts on how to use jsforce before its dependency xml2js is updated with latest version.

**Comment by batflarrow** [comment]
> Hey folks 👋🏼
> 
> We just published a new version on the beta channel (`2.0.0-beta.21`) with xml2js 0.5.0.
> 
> Thanks for the heads up!

Hi there,

I'm wondering if there's any plan to backport this package upgrade to v1.x? Alternatively, is there a migration guide available to help upgrade from v1.x to v2.x? As v2.x is still in beta, it would be helpful to have a list of known caveats to watch out for.

I couldn't find a migration guide or list of known caveats anywhere. Additionally, we experienced random type errors in our existing code when upgrading to this beta channel from v1.11.0. Therefore, either backporting the xml2js upgrade to v1.x or providing some kind of migration guide would be greatly appreciated.

Thank you in advance!

**Comment by cristiand391** [comment]
@batflarrow 
I don't think there's a migration guide yet, you can see this blog post about the first alpha version:
https://jsforce.github.io/blog/posts/20191216-jsforce20-alpha-preview-with-schema-type-feature.html

IIRC, when we fully moved to v2 in the Salesforce CLI last year we had a few type issues but nothing huge.
Still, we agreed that asking users to move to v2 to get the security fix may not be possible for all  and decided we'll patch v1 this week 👍🏼 .

That being said, we are planning a possible GA of v2 so any feedback you have would help a lot (you can create a new issue in this repo and ping me).

**Comment by msouther-lisc** [comment]
I'm excited to hear V2 might finally go GA!

That news alone makes this little security bug well worth it 😋

**Comment by cristiand391** [comment]
@batflarrow just published jsforce v1.11.1 with xml2.js 0.5.0.

**Comment by batflarrow** [comment]
> @batflarrow just published jsforce v1.11.1 with xml2.js 0.5.0.

Thank you so much for this. 
I will upgrade to this for now and surely upgrade to v2 when it finally goes GA, and come back with feedback if any!
